civis/polling.py

```diff
@@ -104,6 +117,64 @@ def __repr__(self):
                                             self._civis_state)
             return out
 
+    def _subscribe(self):
+        pubnub = None
+        if self.poller_args:
+            class JobCompleteListener(SubscribeCallback):
```

Comment:
After thinking about it some more, I think that it might make more sense to leave `PollableResult` alone and create a separate class that implements the same interface for the Pubnub integration. Then we could create a wrapper that would connect to Pubnub if they have the requirements and fallback to Polling if they don't or if the Pubnub connection fails.